x/msgfees/client/cli/tx.go

```diff
@@ -57,90 +53,8 @@ For add, update, and removal of msg fees amount and min fee and/or rate fee must
 $ %[1]s tx msgfees update "updating" "updating MsgWriterRecordRequest fee"  10nhash --msg-type=/provenance.metadata.v1.MsgWriteRecordRequest --additional-fee=612000nhash --recipient=pb... --bips=5000
 $ %[1]s tx msgfees remove "removing" "removing MsgWriterRecordRequest fee" 10nhash --msg-type=/provenance.metadata.v1.MsgWriteRecordRequest
 `, version.AppName),
-		RunE: func(cmd *cobra.Command, args []string) error {
-			clientCtx, err := client.GetClientTxContext(cmd)
-			if err != nil {
-				return err
-			}
-
-			proposalType := args[0]
-
-			msgType, err := cmd.Flags().GetString(FlagMsgType)
-			if err != nil {
-				return err
-			}
-
-			_, err = clientCtx.InterfaceRegistry.Resolve(msgType)
-			if err != nil {
-				return err
-			}
-
-			recipient, err := cmd.Flags().GetString(FlagRecipient)
-			if err != nil {
-				return err
-			}
-
-			bips, err := cmd.Flags().GetString(FlagBips)
-			if err != nil {
-				return err
-			}
-
-			var addFee sdk.Coin
-			if proposalType != "remove" {
-				additionalFee, errMinFee := cmd.Flags().GetString(FlagMinFee)
-				if errMinFee != nil {
-					return err
-				}
-				if additionalFee != "" {
-					addFee, err = sdk.ParseCoinNormalized(additionalFee)
-					if err != nil {
-						return err
-					}
-				}
-			}
-
-			var proposal govtypesv1beta1.Content
-
-			switch args[0] {
-			case "add":
-				proposal = &types.AddMsgFeeProposal{
-					Title:                args[1],
-					Description:          args[2],
-					MsgTypeUrl:           msgType,
-					AdditionalFee:        addFee,
-					Recipient:            recipient,
-					RecipientBasisPoints: bips,
-				}
-			case "update":
-				proposal = &types.UpdateMsgFeeProposal{
-					Title:                args[1],
-					Description:          args[2],
-					MsgTypeUrl:           msgType,
-					AdditionalFee:        addFee,
-					Recipient:            recipient,
-					RecipientBasisPoints: bips,
-				}
-			case "remove":
-				proposal = &types.RemoveMsgFeeProposal{
-					Title:       args[1],
-					Description: args[2],
-					MsgTypeUrl:  msgType,
-				}
-			default:
-				return fmt.Errorf("unknown proposal type %q", args[0])
-			}
-
-			deposit, err := sdk.ParseCoinsNormalized(args[3])
-			if err != nil {
-				return err
-			}
-
-			callerAddr := clientCtx.GetFromAddress()
-			msg, err := govtypesv1beta1.NewMsgSubmitProposal(proposal, deposit, callerAddr)
-			if err != nil {
-				return fmt.Errorf("invalid governance proposal. Error: %w", err)
-			}
-			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), msg)
+		RunE: func(_ *cobra.Command, _ []string) error {
+			return fmt.Errorf("this command has been deprecated, and is no longer functional. Please use 'gov proposal submit-proposal' instead")
```

Comment:
The `tx gov submit-proposal` endpoint is clunky and annoying. Since the content of the proposals and msgs should be pretty much the same, it shouldn't be too tough to tweak it to send the msgs as a gov prop. It's pretty easy. Use `govcli.AddGovPropFlagsToCmd` and `provcli.AddAuthorityFlagToCmd` to add some flags to the command. Then, use `provcli.GetAuthority` to set the `authority` fields, and, instead of `GenerateOrBroadcastTxCLI`, use `provcli.GenerateOrBroadcastTxCLIAsGovProp`.

If you decided to just leave it deprecated, you'll need to fix the message. The command is `gov submit-proposal` (no `proposal`).

This goes for the other commands in here too.

Also, the proposal proto messages should be marked as deprecated too.

Reply:
Ok, yeah I copied and pasted this from another area that we just deprecated the call.  I will fix the error that error message.  However, I will make this one update submit the proposals.